test/dlc_tests/sanity/test_ecr_scan.py

```diff
@@ -181,15 +221,90 @@ def test_ecr_scan(image, ecr_client, sts_client, region):
         else:
             raise RuntimeError(display_message)
+
+
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+@pytest.mark.integration("ECR Enhanced Scans on Images")
+def test_ecr_enhanced_scan(image, ecr_client, sts_client, region):
+    """
+    Run ECR Enhanced Scan Tool on an image being tested, and raise Error if vulnerabilities found
+    1. Upload image to the ECR Enhanced Scanning Testing Repo.
+    2. Wait for the scans to complete - takes approx 10 minutes for big images. Once the scan is complete, 
+        the scan status changes to ACTIVE
+    3. If the status does not turn to ACTIVE, raise a TimeOut Error
+    4. Read the ecr_scan_results and remove the allowlisted vulnerabilities from it
+    5. In case any vulnerability is remaining after removal, raise an error
+
+    :param image: str Image URI for image to be tested
+    :param ecr_client: boto3 Client for ECR
+    :param sts_client: boto3 Client for STS
+    :param region: str Name of region where test is executed
+    """
+    LOGGER.info(f"Running test_ecr_enhanced_scan for image {image}")
+    image = conduct_preprocessing(image, ecr_client, sts_client, region)
+
+    new_uri = get_new_image_uri_using_current_uri_and_new_repo(
+        image,
+        new_repository_name=ECR_ENHANCED_SCANNING_REPO_NAME,
+        new_repository_region=ECR_ENHANCED_REPO_REGION,
+        append_tag="ENHSCAN",
+    )
+
+    run(f"docker tag {image} {new_uri}", hide=True)
+    ecr_utils.reupload_image_to_test_ecr(
+        new_uri, ECR_ENHANCED_SCANNING_REPO_NAME, ECR_ENHANCED_REPO_REGION, pull_image=False
+    )
+
+    ecr_client_for_enhanced_scanning_repo = boto3.client("ecr", region_name=ECR_ENHANCED_REPO_REGION)
+    wait_for_enhanced_scans_to_complete(ecr_client_for_enhanced_scanning_repo, new_uri)
+    sleep(1 * 60)
+
+    scan_results = ecr_utils.get_all_ecr_enhanced_scan_findings(
+        ecr_client=ecr_client_for_enhanced_scanning_repo, image_uri=new_uri
+    )
+    scan_results = json.loads(json.dumps(scan_results, default=ecr_utils.ecr_json_serializer))
+
+    minimum_sev_threshold = get_minimum_sev_threshold_level(image)
+    ecr_image_vulnerability_list = ECREnhancedScanVulnerabilityList(minimum_severity=CVESeverity[minimum_sev_threshold])
+    ecr_image_vulnerability_list.construct_allowlist_from_ecr_scan_result(scan_results)
+
+    LOGGER.info(f"ecr_image_vulnerability_list formed {ecr_image_vulnerability_list.vulnerability_list}")
+
+    remaining_vulnerabilities = ecr_image_vulnerability_list
+
+    image_scan_allowlist = ECREnhancedScanVulnerabilityList(minimum_severity=CVESeverity[minimum_sev_threshold])
+
+    try:
+        image_scan_allowlist_path = get_ecr_scan_allowlist_path(image)
+        if os.path.exists(image_scan_allowlist_path):
+            image_scan_allowlist.construct_allowlist_from_file(image_scan_allowlist_path)
+    except:
+        LOGGER.info(f"[AllowlistPathNotFound] Image scan allowlist path not found for {image}")
```

Comment:
`os.path.exists(image_scan_allowlist_path)` is meant to check the existence of the Dockerfile. However, before we check the existence of the dockerfile, we retrieve the correct name of the Dockerfile by calling `get_ecr_scan_allowlist_path` method, which in turn calls `get_dockerfile_path_for_image` [link](https://github.com/aws/deep-learning-containers/blob/master/test/test_utils/__init__.py#L170) method that raises [LookupError](https://github.com/aws/deep-learning-containers/blob/master/test/test_utils/__init__.py#L224). The except is there to catch those errors and prevent failure (especially since we do not have allowlists right now)